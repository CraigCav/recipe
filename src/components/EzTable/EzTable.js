import styled from 'react-emotion';
import {base} from './EzTable.styles';

/**
 * Tables display information in a way thatâ€™s easy to scan,
 * so that users can look for patterns and insights.
 * They can be embedded in primary content, such as cards.
 */
const EzTable = styled('div')(base);

EzTable.propTypes = {};

/**
 * defaultProps
 */
EzTable.defaultProps = {};

/**
 * @component
 */
export default EzTable;
