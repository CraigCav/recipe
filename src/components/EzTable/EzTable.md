---
name: Table
title: Table
path: '/components/ez-table'
---

Tables are used to display information from a data set in a way that's easy to scan. Tables allow users to compare and analyze the data to look for patterns and insights. Tables will often be the primary content within a [Card](/components/ez-card).

---

<EzAlert
  headline="This component is under development"
  tagline="There will likely be breaking changes to the API. Proceeed with caution."
  use="warning"
/>

<br/>
<br/>

Features still in consideration include:

- Responsive variants (cards with repeated labels, transpose table)
- Column pinning to support horizontal scrolling
- Headerless table
- Sorting
- Filtering
- Row hover styles (interactive)
- Zebra Striping
- Pagination
- Table Actions (print, download etc)
- Column width options (fixed, grow, auto, ellipsis, tooltip)
- Headerless columns (content between columns, e.g. formula operators)
- Sticky headers

---

## Best Practices

Tables should:

- Be used to show homogeneous content.
- Help users visualize values from a data set.
- Only display values supporting the data's purpose.
- Include a heading to help identify the contents of the table at-a-glance.

## Alignment

Column content types can be varied using component props to enable the following alignment rules:

- Numerical values should be right aligned
- Textual values should be left aligned
- Headers (where present) should align with their data

---

## Content guidelines

Tables should:

- Include concise but descriptive headers
- Avoid truncating content where possible

Tables can also:

- Support items that perform an action when clicked. For example, navigating to the item's details page or provide more detail about the item.
- Provide bulk actions for tasks that are applicable to many items at once.

## Examples

### Simple table

Used whenever the tablular data is directly related to the preceeding or subsequent content. Simple tables should be used as a direct child of a [Card](/components/ez-card) component.

```jsxwide
<EzPage>
  <EzCard>
    <EzHeading
      size="3"
    >
      Card Title
    </EzHeading>
    <EzAlert
      headline="Info message"
      tagline="An example of content that is a sibling of the table."
      use="info"
    />
    <EzTable
      columns={[
        {heading: 'From', accessor: 'from'},
        {heading: 'Through', accessor: 'to'},
        {heading: 'Total time closed', accessor: 'total'}
      ]}
      items={[
        {from: '9/3/18', to: '9/5/18', total: '3 days'},
        {from: '10/8/18', to: '10/8/18', total: '1 days'}
      ]} />
    <p>Tables may share a card with other related content.</p>
  </EzCard>
</EzPage>
```

### Tables that fill a Card

Used whenever the tablular data alone represents a cohesive set of content. Should be used with a `title` prop and may include optional actions.

```jsxwide
<EzPage>
  <EzTable
    title="All Stores"
    subtitle="Compared to the same period last year"
    columns={[
      {heading: 'Store name', accessor: 'store'},
      {heading: 'Total sales', accessor: 'total', numeric: true},
      {heading: 'Average order value', accessor: 'average', numeric: true}
    ]}
    items={[
      {id: '#004', store: '123 Example Store', total: 23267, average: 327.79},
      {id: '#007', store: '45 Meadowview Lane', total: 22788, average: 367.55},
    ]} />
</EzPage>
```

### Custom cell rendering

Use when more fine-grained control over the table content is desired (in favor of simple Textual/Numerical content).

```jsxwide
<Component>
  {() => {
    // declare any component to define your custom column template
    const StoreName = ({item: {store, id}}) => (
      <div>
        <div>
          <a href="javascript:void(0);">{store}</a>
        </div>
        <div>
          <EzTextStyle use="subdued">{id}</EzTextStyle>
        </div>
      </div>
    );

    return (
      <EzPage>
        <EzTable
          title="All Stores"
          subtitle="Compared to the same period last year"
          columns={[
            {heading: 'Store name', accessor: StoreName},
            {heading: 'Total sales', accessor: 'total', numeric: true},
            {heading: 'Average order value', accessor: 'average', numeric: true}
          ]}
          items={[
            {id: '#004', store: '123 Example Store', total: 23267, average: 327.79},
            {id: '#007', store: '45 Meadowview Lane', total: 22788, average: 367.55},
          ]} />
      </EzPage>
    );
  }}
</Component>
```

### Bulk row selection

Use when multiple rows of a table can be selected together in order to perform an action in bulk.

Providing the bulk select event handler props (`onRowSelectClick`, `onBulkSelectClick`, `isRowSelected`) will add a column to each table row containing a checkbox input, along with a corresponding column header containing a checkbox input.

The row-level checkbox input can be toggled to indicate that the current row should be included in a particular action. This functionality is handled by the `isRowSelected` and `onRowSelectClick` props. `isRowSelected` is a Function that is used to determine the row-level checkbox input state. It is called when rendering each row, and is passed the row's `item` object as the single argument. `onRowSelectClick` is a Function that is bound to the row-level checkbox input change handler. It is called when the input state changes, and is passed the row's `item` object as the single argument.

The column header checkbox input can be toggled to select or deselect all currently visible rows. This functionality is handled by the `onBulkSelectClick` prop. This should be a Function that handles the behavior of selecting or deselecting all the visible table rows.

The column header checkbox input state and behavior is determined by evaluating the state of each visible table row. If all rows are selected, then the checkbox will appear selected, and deselecting the input should deselect all rows. If some or none of the rows are selected, the checkbox will appear deselected, and selecting the input should select all rows.

```jsxwide
<Component initialState={{selectedStoreIds: []}}>
  {({state, setState}) => {
    const storeIds = ['#001', '#002'];

    const selectAll = () => {
      setState({selectedStoreIds: storeIds});
    }

    const deselectAll = () => {
      setState({selectedStoreIds: []});
    }

    const selectRow = item => {
      const {selectedStoreIds} = state;

      if (!selectedStoreIds.includes(item.id)) {
        const newStoreIds = [...selectedStoreIds];
        newStoreIds.push(item.id);
        setState({selectedStoreIds: newStoreIds});
      }
    }

    const deselectRow = item => {
      const {selectedStoreIds} = state;
      const newStoreIds = [...selectedStoreIds].filter(id => id !== item.id);
      setState({selectedStoreIds: newStoreIds});
    }

    const isRowSelected = item => {
      const {selectedStoreIds} = state;
      return selectedStoreIds.some(id => id === item.id);
    }

    const onBulkSelectClick = () => {
      const {selectedStoreIds} = state;

      selectedStoreIds.length === storeIds.length ? deselectAll() : selectAll();
    }

    const onRowSelectClick = (_event, {item}) => {
      isRowSelected(item) ? deselectRow(item) : selectRow(item);
    }

    return (
      <EzPage>
        <EzTable
          title="All Stores"
          subtitle="Compared to the same period last year"
          onBulkSelectClick={onBulkSelectClick}
          onRowSelectClick={onRowSelectClick}
          isRowSelected={isRowSelected}
          columns={[
            {heading: 'Store name', accessor: 'store'},
            {heading: 'Total sales', accessor: 'total', numeric: true},
            {heading: 'Average order value', accessor: 'average', numeric: true}
          ]}
          items={[
            {id: storeIds[0], store: '123 Example Store', total: 23267, average: 327.79},
            {id: storeIds[1], store: '45 Meadowview Lane', total: 22788, average: 367.55},
          ]} />
      </EzPage>
    );
  }}
</Component>
```
