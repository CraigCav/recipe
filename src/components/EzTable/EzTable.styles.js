import styled, {css} from 'react-emotion';
import {EzCardSection} from '../EzCard';

const fullBleed = css`
  padding: 0;
`;

const spacing = ({
  theme: {
    spacing: {lg},
  },
}) => css`
  th,
  td {
    &:first-child {
      padding-left: ${lg};
    }
    &:last-child {
      padding-right: ${lg};
    }
  }
`;

const cell = ({numeric}) =>
  numeric !== undefined &&
  css`
    text-align: right;
  `;

const heading = ({theme}) => css`
  color: ${theme.colors.grays[600]};
`;

const borders = ({theme}) => css`
  th,
  td {
    border-bottom: 1px solid ${theme.colors.grays[300]};
  }

  tr:last-child td {
    border-bottom: none;
  }
`;

export const Th = styled.th(cell, heading);
export const Td = styled.td(cell);

export const TableCardSection = styled(EzCardSection)(fullBleed, spacing, borders);

// EzCard checks for a card section (by displayName), and will wrap if it can't find one
TableCardSection.displayName = EzCardSection.displayName;

const base = css`
  margin: 0;

  th,
  td {
    border: none;
  }
`;

export const Table = styled.table(base);
