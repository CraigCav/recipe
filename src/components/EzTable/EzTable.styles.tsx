import {css} from 'emotion';
import {EzCardSection} from '../EzCard';
import styled, {Theme} from '../../themes/styled';

const fullBleed = css`
  padding: 0;
`;

const spacing = ({theme}) => css`
  th,
  td {
    &:first-child {
      padding-left: ${theme.spacing.lg};
    }
    &:last-child {
      padding-right: ${theme.spacing.lg};
    }
  }
`;

const interactiveStyles = css`
  cursor: pointer;
  user-select: none;
`;

const cell = ({theme, numeric}) => css`
  text-align: ${numeric ? 'right' : 'left'};
  padding: ${theme.spacing.sm} ${theme.spacing.xs};
`;

const heading = ({theme}) => css`
  color: ${theme.colors.grays[600]};
`;

const borders = ({theme}) => css`
  tbody tr:first-child td {
    border-top: 1px solid ${theme.colors.grays[300]};
  }

  && td {
    border-bottom: 1px solid ${theme.colors.grays[300]};
  }

  tr:last-child td {
    border-bottom: none;
  }
`;

const sortable = ({theme, sorted, sortable: isSortable}) =>
  isSortable &&
  css`
    ${interactiveStyles};

    span {
      display: inline-flex;
      align-items: center;
    }
    svg {
      fill: ${theme.colors.grays[600]};
      margin-left: ${theme.spacing.xs};
      opacity: ${sorted ? '1' : '0'};
    }
  `;

type CellProps = {
  numeric?: boolean;
  sorted?: boolean;
};

const rowHover = ({clickable, theme}) =>
  clickable &&
  css`
    &&:hover {
      td {
        background-color: ${theme.colors.blues[100]};
        user-select: none;
        cursor: pointer;
      }
    }
  `;

export const Th = styled.th<CellProps & {sortable?: boolean}>(cell, heading, sortable);
export const Td = styled.td<CellProps>(cell);
export const ClickableTr = styled.tr<{clickable: boolean}>(rowHover);

type TableCardSectionProps = {
  theme?: Theme;
};

export const TableCardSection = styled(EzCardSection)<TableCardSectionProps>(
  fullBleed,
  spacing,
  borders
);

export const TablePaginationNavItem = styled.span`
  padding-right: ${props => props.theme.spacing.lg};
`;

// `direction: rtl` on the select is used to align the text in the select box to the right, changing direction in the option tag prevents the text contents from displaying RTL.
export const TablePaginationRowCountDropdown = styled.select`
  padding-right: ${props => props.theme.spacing.md};
  font-family: Lato, 'Helvetica Neue', Arial, Helvetica, sans-serif;
  font-size: ${prop => prop.theme.fontSizes[300]};
  font-weight: bold;
  color: ${props => props.theme.colors.primary.main};
  border: 0;
  background-image: url('data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iOSIgaGVpZ2h0PSI2IiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHhtbG5zOnhsaW5rPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5L3hsaW5rIj48ZGVmcz48cGF0aCBkPSJNMTIyLjI0MiAxMC40MTVsMy40MzYgMy40MzZjLjAwMy4wMDQuMDA0LjAwOS4wMDguMDEzYS44MjcuODI3IDAgMCAwIC41ODkuMjQxYy4yMTMgMCAuNDI3LS4wOC41ODktLjI0MS4wMDQtLjAwNC4wMDUtLjAxLjAwOC0uMDEzbDMuNDM2LTMuNDM2YS44MzIuODMyIDAgMCAwIDAtMS4xNzMuODMyLjgzMiAwIDAgMC0xLjE3MyAwbC0yLjg2IDIuODYtMi44Ni0yLjg2YS44MzIuODMyIDAgMCAwLTEuMTczIDAgLjgzMi44MzIgMCAwIDAgMCAxLjE3M3oiIGlkPSJhIi8+PC9kZWZzPjx1c2UgZmlsbD0iIzNFOTBENiIgZmlsbC1ydWxlPSJub256ZXJvIiB4bGluazpocmVmPSIjYSIgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoLTEyMiAtOSkiLz48L3N2Zz4=');
  background-repeat: no-repeat;
  background-position: right 60%;
  background-color: transparent;
  appearance: none;
  cursor: pointer;
  direction: rtl;
  option {
    direction: ltr;
  }
`;

/*
  With table-layout: auto, widths of table cells only shrink to the size of their content,
  so setting the width to a size smaller than the content size causes the cell to always fit
  the content and not stretch to fill the available space.
*/
const cellFitContent = '1%';

type Selectable = {selectable?: boolean};

const selectionColumn = ({theme, selectable}) =>
  selectable &&
  css`
    th:first-child,
    td:first-child {
      width: ${cellFitContent};
    }

    && thead tr + tr {
      td {
        background-color: ${theme.colors.blues[200]};
        border-bottom: solid 1px ${theme.colors.blues[400]};
        border-top: solid 1px ${theme.colors.blues[400]};
        padding: ${theme.spacing.xs};

        > * {
          justify-content: center;
        }
      }
    }
  `;

// EzCard checks for a card section (by displayName), and will wrap if it can't find one
TableCardSection.displayName = EzCardSection.displayName;

const base = ({theme}) => css`
  margin: 0;
  line-height: 1.5rem;
  border-collapse: collapse;
  width: 100%;

  th,
  td {
    border: none;
  }

  tr:hover {
    td {
      background-color: ${theme.colors.grays[100]};
    }
  }
`;

export const Table = styled.table<Selectable>(base, selectionColumn);
