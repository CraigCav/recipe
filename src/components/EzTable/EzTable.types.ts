type Column = (SortEnabledColumn | SortDisabledColumn) & {
  heading: string;
  numeric?: boolean;
  defaultSort?: Direction;
};

type SortEnabledColumn = {
  sortable: boolean;
  accessor: string;
};

type SortDisabledColumn = {
  sortable?: never;
  accessor: React.ReactNode | React.ComponentType;
};

type Direction = 'asc' | 'desc';

type OnSortClickOptions = {
  column: Column;
  direction: Direction;
};

type onSortClick = (event: React.MouseEvent<HTMLElement>, options: OnSortClickOptions) => void;

type onRowsPerPageChange = (event: any) => void;

type BulkSelectionDisabled = {
  onBulkSelectClick?: never;
  onRowSelectClick?: never;
  isRowSelected?: never;
};

type BulkSelectionEnabled = {
  onRowSelectClick: (event: React.MouseEvent<HTMLInputElement>, value: any) => void;
  onBulkSelectClick: React.MouseEventHandler;
  isRowSelected: (item: any) => boolean;
};

type SelectAllOrNoneEnabled = {
  onSelectAllClick: React.MouseEventHandler;
  onSelectNoneClick: React.MouseEventHandler;
};

type Selection = {selection?: BulkSelectionEnabled | BulkSelectionDisabled};
type SelectionAndPagination = {selection: BulkSelectionEnabled & SelectAllOrNoneEnabled};

type ActionsProps = {
  actions: React.ReactNode;
  title: string;
};

type OptionalTitle = {
  actions?: never;
  title?: string;
};

type TableActions = ActionsProps | OptionalTitle;

type PaginationProps = {
  currentPage: number;
  totalRows: number;
  rowsPerPage: number;
  rowsPerPageOptions: number[];
  onPrevPageClick: React.MouseEventHandler;
  onNextPageClick: React.MouseEventHandler;
  onRowsPerPageChange: onRowsPerPageChange;
};

type TableBase = {
  subtitle?: string;
  columns: Column[];
  items: any[];
  onSortClick?: onSortClick;
  pagination?: PaginationProps;
};

export type TableProps = TableBase & (Selection | SelectionAndPagination) & TableActions;

export type Sortable = {
  direction: Direction;
  isSorted: (column: Column) => boolean;
  onClick: (
    event: React.MouseEvent<HTMLElement>,
    column: Column,
    callback: (event: React.MouseEvent<HTMLElement>, options: OnSortClickOptions) => void
  ) => void;
};

export type SortingState = {
  direction: Direction;
  column?: Column;
};
