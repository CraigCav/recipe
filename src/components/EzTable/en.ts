export default {
  'Previous Page': 'Previous Page',
  'Next Page': 'Next Page',
  'Page {{currentPage}} of {{pages}}': 'Page {{currentPage}} of {{pages}}',
  '{{num}} rows per page': '{{num}} rows per page',
  'All {{selectedCount}} rows on this page are selected.':
    'All {{selectedCount}} rows on this page are selected.',
  'All {{totalRowCount}} rows are selected.': 'All {{totalRowCount}} rows are selected.',
  'Clear selection': 'Clear selection',
  'Select all {{totalRowCount}} rows': 'Select all {{totalRowCount}} rows',
};
