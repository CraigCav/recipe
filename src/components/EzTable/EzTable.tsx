import React from 'react';
import {EzCard} from '..';
import {Table, Th, Td, TableCardSection} from './EzTable.styles';

const Thead = ({columns}) => (
  <thead>
    <tr>
      {columns.map(({heading, numeric}, cellIndex) => (
        <Th key={cellIndex} numeric={numeric}>
          {heading}
        </Th>
      ))}
    </tr>
  </thead>
);

const Tbody = ({columns, items}) => (
  <tbody>
    {items.map((item, rowIndex) => (
      <tr key={rowIndex}>
        {columns.map(({accessor, numeric}, cellIndex) => (
          <Td key={cellIndex} numeric={numeric}>
            {typeof accessor === 'function' ? accessor({item}) : item[accessor]}
          </Td>
        ))}
      </tr>
    ))}
  </tbody>
);

type Column = {
  heading: string;
  accessor: any;
  numeric?: boolean;
};

type TableProps = {
  title?: string;
  subtitle?: string;
  columns: Column[];
  items: any[];
};

/**
 * Tables display information in a way thatâ€™s easy to scan,
 * so that users can look for patterns and insights.
 * They can be embedded in primary content, such as cards.
 */
const EzTable: React.SFC<TableProps> = ({title, subtitle, columns, items}) => {
  const table = (
    <Table>
      <Thead columns={columns} />
      <Tbody columns={columns} items={items} />
    </Table>
  );

  if (title) {
    return (
      <EzCard title={title} subtitle={subtitle}>
        <TableCardSection>{table}</TableCardSection>
      </EzCard>
    );
  }

  return table;
};

export default EzTable;
