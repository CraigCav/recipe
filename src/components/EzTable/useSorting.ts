import React, {useState} from 'react';
import {Direction, SortingState, Sortable} from './EzTable.types';

const {Ascending, Descending} = Direction;

const useSorting = (
  initialState: SortingState = {direction: Ascending, column: null}
): Sortable => {
  const [state, setState] = useState<SortingState>(initialState);

  const isSorted = React.useCallback(
    column => state.column && state.column.accessor === column.accessor,
    [state]
  );

  const onClick = React.useCallback(
    (event, column, callback) => {
      if (!column.sortable) return;

      const direction = isSorted(column) && state.direction === Ascending ? Descending : Ascending;

      callback(event, {column, direction});

      if (event.isDefaultPrevented()) return;

      setState({column, direction});
    },
    [state, isSorted]
  );

  return {
    direction: state.direction,
    isSorted,
    onClick,
  };
};

export default useSorting;
