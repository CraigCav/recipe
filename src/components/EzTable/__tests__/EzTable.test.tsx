import React from 'react';
import {axe} from 'jest-axe';
import {visualSnapshots} from 'sosia';
import Component from 'react-component-component';
import markdown from '../EzTable.md';
import EzTable from '../EzTable';
import {EzPage, EzCard, EzHeading, EzAlert, EzTextStyle} from '../../index';
import {renderToHtml} from '../../../jest-globals';

const scope = {EzTable, EzPage, EzCard, EzHeading, EzAlert, EzTextStyle, Component};

describe('EzTable', () => {
  visualSnapshots({markdown, scope});

  const columns = [
    {heading: 'Store name', accessor: ({item}) => item.store},
    {heading: 'Total sales', accessor: 'total', numeric: true},
    {heading: 'Average order value', accessor: 'average', numeric: true},
  ];
  const items = [
    {id: '#004', store: '123 Example Store', total: 23267, average: 327.79},
    {id: '#007', store: '45 Meadowview Lane', total: 22788, average: 367.55},
  ];

  it('should meet accessibility guidelines', async () => {
    const wrapper = renderToHtml(<EzTable columns={columns} items={items} />);
    const actual = await axe(wrapper);
    expect(actual).toHaveNoViolations();
  });
});
