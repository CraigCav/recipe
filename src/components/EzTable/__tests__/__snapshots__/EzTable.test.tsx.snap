// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EzTable should render with default styles 1`] = `
.recipe-18 {
  margin: 0;
  line-height: 1.5rem;
  border-collapse: collapse;
  width: 100%;
}

.recipe-18 th,
.recipe-18 td {
  border: none;
}

.recipe-6 {
  text-align: left;
  padding: 0.8571428571428571rem 0.5714285714285714rem;
}

.recipe-0 {
  text-align: left;
  padding: 0.8571428571428571rem 0.5714285714285714rem;
  color: #8b99a6;
}

.recipe-8 {
  text-align: right;
  padding: 0.8571428571428571rem 0.5714285714285714rem;
}

.recipe-2 {
  text-align: right;
  padding: 0.8571428571428571rem 0.5714285714285714rem;
  color: #8b99a6;
}

<table
  className="recipe-18 recipe-19"
>
  <thead>
    <tr>
      <th
        className="recipe-0 recipe-1"
      >
        Store name
      </th>
      <th
        className="recipe-2 recipe-1"
      >
        Total sales
      </th>
      <th
        className="recipe-2 recipe-1"
      >
        Average order value
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td
        className="recipe-6 recipe-7"
      >
        123 Example Store
      </td>
      <td
        className="recipe-8 recipe-7"
      >
        23267
      </td>
      <td
        className="recipe-8 recipe-7"
      >
        327.79
      </td>
    </tr>
    <tr>
      <td
        className="recipe-6 recipe-7"
      >
        45 Meadowview Lane
      </td>
      <td
        className="recipe-8 recipe-7"
      >
        22788
      </td>
      <td
        className="recipe-8 recipe-7"
      >
        367.55
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`EzTable should render with full bleed styles 1`] = `
.recipe-22 {
  margin: 0;
  line-height: 1.5rem;
  border-collapse: collapse;
  width: 100%;
}

.recipe-22 th,
.recipe-22 td {
  border: none;
}

.recipe-10 {
  text-align: left;
  padding: 0.8571428571428571rem 0.5714285714285714rem;
}

.recipe-4 {
  text-align: left;
  padding: 0.8571428571428571rem 0.5714285714285714rem;
  color: #8b99a6;
}

.recipe-12 {
  text-align: right;
  padding: 0.8571428571428571rem 0.5714285714285714rem;
}

.recipe-6 {
  text-align: right;
  padding: 0.8571428571428571rem 0.5714285714285714rem;
  color: #8b99a6;
}

.recipe-28 {
  background: #ffffff;
  border: 1px solid #ced4d9;
  border-radius: 6px;
}

.recipe-2 {
  margin: 1.4285714285714286rem 1.4285714285714286rem 0;
}

.recipe-0 {
  font-size: 1.2857142857142858rem;
  margin: 0;
}

.recipe-26 > *:not(:first-child) {
  border-top: 1px solid #ced4d9;
}

.recipe-24 {
  padding: 1.4285714285714286rem;
  padding: 0;
}

.recipe-24 > *:not(:first-child) {
  margin-top: 1.1428571428571428rem;
}

.recipe-24 th:first-child,
.recipe-24 td:first-child {
  padding-left: 1.4285714285714286rem;
}

.recipe-24 th:last-child,
.recipe-24 td:last-child {
  padding-right: 1.4285714285714286rem;
}

.recipe-24 th,
.recipe-24 td {
  border-bottom: 1px solid #e6e9eb;
}

.recipe-24 tr:last-child td {
  border-bottom: none;
}

<div
  className="recipe-28 recipe-29"
>
  <div
    className="recipe-2 recipe-3"
  >
    <h3
      className="recipe-0 recipe-1"
    >
      My Table
    </h3>
  </div>
  <div
    className="recipe-26 recipe-27"
  >
    <div
      className="recipe-24 recipe-25"
    >
      <table
        className="recipe-22 recipe-23"
      >
        <thead>
          <tr>
            <th
              className="recipe-4 recipe-5"
            >
              Store name
            </th>
            <th
              className="recipe-6 recipe-5"
            >
              Total sales
            </th>
            <th
              className="recipe-6 recipe-5"
            >
              Average order value
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              className="recipe-10 recipe-11"
            >
              123 Example Store
            </td>
            <td
              className="recipe-12 recipe-11"
            >
              23267
            </td>
            <td
              className="recipe-12 recipe-11"
            >
              327.79
            </td>
          </tr>
          <tr>
            <td
              className="recipe-10 recipe-11"
            >
              45 Meadowview Lane
            </td>
            <td
              className="recipe-12 recipe-11"
            >
              22788
            </td>
            <td
              className="recipe-12 recipe-11"
            >
              367.55
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
